src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -0,0 +1,260 @@
+                    if (wantedNode != null)
+                    {
+                        return wantedNode;
+                    }
+
+                    leftNode = leftNode?.Parent;
+                }
+                while (leftNode != null && leftNode.Span.End == selection.Start);
+            }
+
+            // nothing found -> return null
+            return null;
+
+            static SyntaxNode TryGetAcceptedNodeOrExtracted(SyntaxNode node, Predicate<SyntaxNode> predicate, Func<SyntaxNode, ISyntaxFactsService, SyntaxNode> extractNode, ISyntaxFactsService syntaxFacts)
+            {
+                if (node == null)
+                {
+                    return null;
+                }
+
+                if (predicate(node))
+                {
+                    return node;
+                }
+
+
+                var extractedNode = extractNode(node, syntaxFacts);
+                return (extractedNode != null && predicate(extractedNode))
+                    ? extractedNode
+                    : null;
+            }
+        }
+
+        /// <summary>
+        /// <para>
+        /// Extractor function for <see cref="TryGetSelectedNodeAsync(Document, TextSpan, Predicate{SyntaxNode}, Func{SyntaxNode, ISyntaxFactsService, SyntaxNode}, CancellationToken)"/> methods 
+        /// that retrieves expressions from  declarations and assignments. Otherwise returns unchanged <paramref name="node"/>.
+        /// </para>
+        /// <para>
+        /// The rationale is that when user selects e.g. entire local delaration statement [|var a = b;|] it is reasonable
+        /// to provide refactoring for `b` node. Similarly for other types of refactorings.
+        /// </para>
+        /// </summary>
+        protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsService syntaxFacts)
+        {
+            // var a = b;
+            // -> b
+            if (syntaxFacts.IsLocalDeclarationStatement(node))
+            {
+                var variables = syntaxFacts.GetVariablesOfLocalDeclarationStatement(node);
+                if (variables.Count == 1)
+                {
+                    var declaredVariable = variables.First();
+                    var initializer = syntaxFacts.GetInitializerOfVariableDeclarator(declaredVariable);
+
+                    if (initializer != default)
+                    {
+                        var value = syntaxFacts.GetValueOfEqualsValueClause(initializer);
+                        if (value != default)
+                        {
+                            return value;
+                        }
+                    }
+                }
+            }
+
+            // a = b;
+            // -> b
+            if (syntaxFacts.IsSimpleAssignmentStatement(node))
+            {
+                syntaxFacts.GetPartsOfAssignmentExpressionOrStatement(node, out _, out _, out var rightSide);
+                return rightSide;
+            }
+
+            return node;
```

Comment:
nit: i would explicitly put a comment here saying this set of checks is not meant to be exhaustive, and can be augmented in the future if we discover more cases that we think are relevant.  this lets people know in the future that it wasn't intentional that we didnt' have a node, but more that this is a living heuristic that is intended to adapt to our needs as we discover them.